src/component.cpp

```diff
@@ -202,5 +214,58 @@ bool Component::hasVariable(const std::string &name) const
     return mPimpl->findVariable(name) != mPimpl->mVariables.end();
 }
 
+void Component::addReset(const ResetPtr &r)
+{
+    mPimpl->mResets.push_back(r);
+}
+
+bool Component::removeReset(size_t index)
+{
+    bool status = false;
+    if (index < mPimpl->mResets.size()) {
+        mPimpl->mResets.erase(mPimpl->mResets.begin() + index);
+        status = true;
+    }
+
+    return status;
+}
+
+bool Component::removeReset(const ResetPtr &reset)
+{
+    bool status = false;
+    auto result = mPimpl->findReset(reset);
+    if (result != mPimpl->mResets.end()) {
+        mPimpl->mResets.erase(result);
+        status = true;
+    }
+
+    return status;
+}
+
+void Component::removeAllResets()
+{
+    mPimpl->mResets.clear();
+}
+
+ResetPtr Component::getReset(size_t index) const
+{
+    ResetPtr reset = nullptr;
+    if (index < mPimpl->mResets.size()) {
+        reset = mPimpl->mResets.at(index);
+    }
+
+    return reset;
```

Comment:
Are we happy to return null instead of using exceptions?

Reply:
We are not raising any exceptions in libCellML directly.  I think it is something you avoid unless you find yourself in a truly exceptional circumstance.